From 66fb9051130fa1bee02ad2cfcf77de76050ebdb6 Mon Sep 17 00:00:00 2001
From: Maxime Boissonneault <maxime.boissonneault@calculquebec.ca>
Date: Mon, 7 Jul 2025 21:05:48 +0000
Subject: [PATCH 1/2] Update 3 files

- /taguette/main.py
- /taguette/templates/base.html
- /taguette/web/__init__.py
---
 taguette/main.py             | 3 +++
 taguette/templates/base.html | 4 ++--
 taguette/web/__init__.py     | 1 +
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/pyproject.toml b/pyproject.toml
index bab1fbe..bb291b7 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -63,6 +63,7 @@ SQLAlchemy = ">=1.4,<1.5"
 subtitle-parser = ">=1,<2"
 tornado = ">=6.1"
 XlsxWriter = ">=1.4,<4"
+setuptools = "<=80.9.0"

 opentelemetry-distro = {version = "*", optional = true, allow-prereleases = true}
 opentelemetry-instrumentation-sqlalchemy = {version = ">=0.24b0,<0.26", optional = true, allow-prereleases = true}

diff --git a/taguette/main.py b/taguette/main.py
index 6aa01237..ad4bf7e4 100644
--- a/taguette/main.py
+++ b/taguette/main.py
@@ -325,6 +325,8 @@ def main():
     parser.add_argument('--dont-set-umask', action='store_const', dest='umask',
                         const=None,
                         help=_("Don't change umask on startup"))
+    parser.add_argument('--base_path', default='/',
+                        help=_("Base path for the URL"))
     parser.set_defaults(
         func=None,  # Function to run instead of default, e.g. migrate, server
         func1=None,  # Run immediately, e.g. before anything is logged
@@ -423,6 +425,7 @@ def main():
             SECRET_KEY=secret,
             COOKIES_PROMPT=False,
             HTML_OUT_SIZE_LIMIT=5000000,  # 5 MB
+            BASE_PATH=args.base_path
         )
 
     if 'PROMETHEUS_LISTEN' in config:
diff --git a/taguette/templates/base.html b/taguette/templates/base.html
index 0249dd8c..349062d2 100644
--- a/taguette/templates/base.html
+++ b/taguette/templates/base.html
@@ -20,8 +20,8 @@
     <nav class="navbar navbar-expand-lg navbar-dark bg-primary">
       <div class="container">
         <a class="navbar-brand special-brand" href="{{ reverse_url('index') }}">
-            <img src="/static/logo.png" alt="LibreQDA" class="not-scrolled">
-            <img src="/static/logo-h.png" alt="LibreQDA" class="scrolled">
+            <img src="{{ static_url('logo.png') }}" alt="LibreQDA" class="not-scrolled">
+            <img src="{{ static_url('logo-h.png') }}" alt="LibreQDA" class="scrolled">
         </a>
         <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
           <span class="navbar-toggler-icon"></span>
diff --git a/taguette/web/__init__.py b/taguette/web/__init__.py
index 62a958e9..e8a48f12 100644
--- a/taguette/web/__init__.py
+++ b/taguette/web/__init__.py
@@ -218,6 +218,7 @@ def make_app(config, debug=False, xsrf_cookies=True):
     return Application(
         routes,
         static_path=pkg_resources.resource_filename('taguette', 'static'),
+        static_url_prefix=base_path + "/static",
         login_url=base_path + '/login',
         xsrf_cookies=xsrf_cookies,
         debug=debug,
-- 
GitLab


From 32992e1b9a5e200f9671c33d7a6204f016b414dc Mon Sep 17 00:00:00 2001
From: Maxime Boissonneault <maxime.boissonneault@calculquebec.ca>
Date: Mon, 7 Jul 2025 21:37:49 +0000
Subject: [PATCH 2/2] Update 2 files

- /taguette/main.py
- /taguette/web/base.py
---
 taguette/main.py     | 30 ++++++++++++++++++++++--------
 taguette/web/base.py |  3 +++
 2 files changed, 25 insertions(+), 8 deletions(-)

diff --git a/taguette/main.py b/taguette/main.py
index ad4bf7e4..9d2f0dfa 100644
--- a/taguette/main.py
+++ b/taguette/main.py
@@ -13,6 +13,8 @@ import string
 import subprocess
 import sys
 import tornado.ioloop
+from tornado.httpserver import HTTPServer
+from tornado.netutil import bind_unix_socket
 from urllib.parse import urlparse
 import webbrowser
 
@@ -327,6 +329,8 @@ def main():
                         help=_("Don't change umask on startup"))
     parser.add_argument('--base_path', default='/',
                         help=_("Base path for the URL"))
+    parser.add_argument('--unix_socket', default=None,
+                        help=_("Unix socket to bind to instead of port"))
     parser.set_defaults(
         func=None,  # Function to run instead of default, e.g. migrate, server
         func1=None,  # Run immediately, e.g. before anything is logged
@@ -425,7 +429,8 @@ def main():
             SECRET_KEY=secret,
             COOKIES_PROMPT=False,
             HTML_OUT_SIZE_LIMIT=5000000,  # 5 MB
-            BASE_PATH=args.base_path
+            BASE_PATH=args.base_path,
+            UNIX_SOCKET=args.unix_socket,
         )
 
     if 'PROMETHEUS_LISTEN' in config:
@@ -455,8 +460,13 @@ def main():
     if config['ALLOW_EXTERNAL_LOGIN_BYPASS']:
         xsrf_cookies = False
     app = make_app(config, debug=args.debug, xsrf_cookies=xsrf_cookies)
-    app.listen(config['PORT'], address=config['BIND_ADDRESS'],
-               xheaders=config.get('X_HEADERS', False))
+    if args.unix_socket:
+        server = HTTPServer(app)
+        socket = bind_unix_socket(args.unix_socket)
+        server.add_socket(socket)
+    else:
+        app.listen(config['PORT'], address=config['BIND_ADDRESS'],
+                   xheaders=config.get('X_HEADERS', False))
     loop = tornado.ioloop.IOLoop.current()
 
     db = app.DBSession()
@@ -477,10 +487,14 @@ def main():
         logger.warning("Debug mode is ON")
         asyncio.get_event_loop().set_debug(True)
 
-    url = get_join_url(app)
-    print(_("\n    Taguette %(version)s is now running. You can connect to it "
-            "using this link:\n\n    %(url)s\n") %
-          dict(url=url, version=exact_version()), flush=True)
+    if args.unix_socket:
+        print(_("\n    Taguette %(version)s is now running using UNIX socket %(socket)s") %
+              dict(socket=args.unix_socket, version=exact_version()), flush=True)
+    else:
+        url = get_join_url(app)
+        print(_("\n    Taguette %(version)s is now running. You can connect to it "
+                "using this link:\n\n    %(url)s\n") %
+              dict(url=url, version=exact_version()), flush=True)
 
     if args.browser and not args.debug:
         loop.call_later(0.01, webbrowser.open, url)
@@ -489,4 +503,4 @@ def main():
 
 
 if __name__ == '__main__':
-    main()
+    main()
\ No newline at end of file
diff --git a/taguette/web/base.py b/taguette/web/base.py
index 57c01271..7077a25c 100644
--- a/taguette/web/base.py
+++ b/taguette/web/base.py
@@ -184,6 +184,9 @@ class Application(GracefulExitApplication):
             logger.info("Starting in multi-user mode")
             if not self.terms_of_service:
                 logger.warning("No terms of service set")
+        elif config['UNIX_SOCKET']:
+            self.single_user_token = None
+            logger.info("Starting with unix socket")
         else:
             self.single_user_token = hmac.new(
                 cookie_secret.encode('utf-8'),
-- 
GitLab
diff --git a/taguette/web/views.py b/taguette/web/views.py
--- a/taguette/web/views.py	2025-06-17 15:15:46.000000000 +0000
+++ b/taguette/web/views.py	2025-07-10 21:04:27.118697751 +0000
@@ -112,7 +112,7 @@
             )
         elif not self.application.config['MULTIUSER']:
             token = self.get_query_argument('token', None)
-            if token and token == self.application.single_user_token:
+            if self.application.config['UNIX_SOCKET'] or (token and token == self.application.single_user_token):
                 self.login('admin')
                 return self.redirect(self.reverse_url('index'))
             elif token:
